engine/table/src/main/java/io/deephaven/engine/table/impl/sources/UnionSourceManager.java

```diff
@@ -411,13 +436,23 @@ else if (nextCurrentKey == nextModifiedKey) {
                 }
                 // Existing constituent processing
                 else {
-                    processExisting(nextCurrentValue);
+                    try {
+                        processExisting(nextCurrentValue);
+                    } catch (ConstituentTableErrorException ex) {
+                        constituentExceptions = collectConstituentException(constituentExceptions, ex);
+                    }
                     advanceCurrent();
                     ++nextCurrentSlot;
                     ++nextPreviousSlot;
                 }
             }
 
+            if (constituentExceptions != null) {
+                throw new UncheckedDeephavenException(
```

Comment:
This is bad. You're throwing to L350 in `MergedListener`, which is going to add some nonsense about uncaught exceptions.
You want to make `MergedListener.propagateError` `protected` and call that. Don't wrap in an `UncheckedDeephavenException` either.